src/main/java/uk/gov/ida/matchingserviceadapter/validators/validationrules/ConditionsShouldNotContainProxyRestrictionElement.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.ida.matchingserviceadapter.validators.validationrules;
+
+import org.opensaml.saml.saml2.core.ProxyRestriction;
+import uk.gov.ida.matchingserviceadapter.exceptions.SamlResponseValidationException;
+
+public class ConditionsShouldNotContainProxyRestrictionElement extends ValidationRule<ProxyRestriction> {
+    public ConditionsShouldNotContainProxyRestrictionElement() {
+        super((e) -> e == null);
```

Comment:
Same for this one as your comment above maybe? `e.getProxyRestriction() == null`

Reply:
Just a thought, is it worth changing `e` in all these validation rules to something else? I see `e` and I think `error`. Maybe `c`, or `conditions`?

Reply:
That's reasonable.  I think I had in mind 'e'lement but it is confusable with error or exception.